Add nilup, nada and instrumentation doc (#13)
NillionNetwork/nillion-docs: docs/nada.md

### Generate a test file
To be able to `run` or `test` programs we need a test file with the inputs and expected outputs.
`nada` have the ability to generate a test file with example values for you.
To generate a test file you can run
```
nada generate-test --test-name <test-name> <program-name>
```
It will generate a test file with the name `<test-name>.yaml` in the `tests` directory.

You should edit the test file to change the inputs and the expected outputs.

### Run a program
To `run` a program using nada you need a test file, goto the [Generate a test file](#Generate-a-test-file) section to see how to generate it.

#### Run in normal mode
To `run` a program you can run
```
nada run <test-name>
```
It will run the program associated to that test file and will print the output

#### Run in debug mode
`nada` have the ability to run a program in debug mode, 
in this mode the operations are done in plain, so you can inspect the intermediary values.
When running in debug mode you can see the operations that are done in the program with the corresponding values.

To run a program in debug you can run
```
nada run --debug <test-name>
```

> It will run the program associated to that test file and will print all operations, values and the output

Reviewer: ```suggestion
This will run the program associated with that test file and print all operations, values and the output
```

### Test a program

> To `test` a program using `nada` you need a test file, goto the [Generate a test file](#Generate-a-test-file) section to see how to generate it.

Reviewer: ```suggestion
To `test` a program using `nada` you need a test file, visit the [Generate a test file](#Generate-a-test-file) section to see how to generate it.
```

#### Test in normal mode
To run all the tests in the project you can run
```
nada test
```
To run a single test you can run
```
nada test <test-name>
```

> It will run the program associated to that test file and will check if the output is the expected one.

Reviewer: ```suggestion
This will run the program associated with that test file and check if the output matches what's expected.
```

#### Test in debug mode

> You can also run a test in debug mode, to know more about debug mode go to [Run in debug mode](#Run-in-debug-mode) section.

Reviewer: ```suggestion
You can also run a test in debug mode, to know more about debug mode visit the [Run in debug mode](#Run-in-debug-mode) section.
```
